src/tests/integration/character-countdown.test.ts

```diff
@@ -0,0 +1,154 @@
+    before(() => {
+      mocks.backgroundWindow.store.dispatchers.updateEditorState({
+        editorState: fromText(almost280Chars),
+      })
+    })
+
+    it('renders a more full progress circle', () => {
+      // TODO: Hank to write this test
+      const characterCountdown = mocks.app().querySelector('.character-countdown')! as HTMLDivElement
+      const progressCircle = characterCountdown.querySelector('circle.progress')! as SVGCircleElement
+      const remainingRatio = Number(progressCircle.style.strokeDashoffset) / Number(progressCircle.style.strokeDasharray)
+      const numCharacters = getPlainText(ensureActiveTab(mocks.getState()).feedbackState.editorState).length
+      const filledInRatio = numCharacters / 280
+      expect(remainingRatio).to.closeTo(0, 0.1)
+      expect(filledInRatio).to.closeTo(1, 0.1)
+    })
+
+    it('has a .warning class', () => {
+      // TODO: Hank to write this test
+      const characterCountdown = mocks.app().querySelector('.character-countdown')! as HTMLDivElement
+      expect(characterCountdown.querySelectorAll('svg.warning')).to.have.length(1)
+    })
+
+    it('renders the number of characters remaining as a text element', () => {
+      // TODO: Hank to write this test
+      expect(mocks.app().querySelectorAll('.warning > text')).to.have.length(1)
+    })
+
+    it('does not disable posting', () => {
+      const postButton = mocks.app().querySelector('.post-btn')! as HTMLButtonElement
+      expect(postButton.disabled).to.equal(false)
+    })
+  })
+
+  describe('when exactly at the character limit', () => {
+    const exactly280Chars = '@zing.com ' + 'x'.repeat(270)
+    before(() => {
+      mocks.backgroundWindow.store.dispatchers.updateEditorState({
+        editorState: fromText(exactly280Chars),
+      })
+    })
+
+    it('renders a full progress circle', () => {
+      const progressCircle = mocks.app().querySelector('.character-countdown circle.progress')! as SVGCircleElement
+      const remainingRatio = Number(progressCircle.style.strokeDashoffset) / Number(progressCircle.style.strokeDasharray)
+      expect(remainingRatio).to.equal(0)
+    })
+
+    it('has an aria-valuenow equal to 100', () => {
+      // TODO: Hank to write this test
+      const characterCountdown = mocks.app().querySelector('.character-countdown')! as HTMLDivElement
+      expect(Number(characterCountdown.getAttribute('aria-valuenow'))).to.equal(100)
+    })
+
+    it('has a .warning class', () => {
+      // TODO: Hank to write this test
+      const characterCountdown = mocks.app().querySelector('.character-countdown')! as HTMLDivElement
+      expect(characterCountdown.querySelectorAll('svg.warning')).to.have.length(1)
+    })
+
+    it('renders zero as the number of characters remaining in a text element', () => {
+      // TODO: Hank to write this test
+      const characterCountdown = mocks.app().querySelector('.character-countdown')! as HTMLDivElement
```

Comment:
Nice job! We can tidy this up a bit using the postfix `!` operator to tell typescript we know the text exists

```typescript
const text = mocks.app().querySelector('.character-countdown text')!
expect(text.innerHTML).to.equal('0')
```

Reply:
resolved. thanks